File: java8/src/main/java/com/lambdacult/spark/example/java8/Handler.java
package com.lambdacult.spark.example.java8;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import org.apache.log4j.Logger;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Map;
import java.util.stream.Collectors;

public class Handler implements RequestHandler<Map<String, Object>, Object> {
    private static final Logger LOG = Logger.getLogger(Handler.class);

    @Override
    public Object handleRequest(Map<String, Object> input, Context context) {
        LOG.info("received: " + input);
        return buildResponseBody(input, context);
        // Or as lambda-proxy response for API Gateway:
        // return ApiGatewayResponse.builder().setStatusCode(200).setObjectBody(buildResponseBody(input, context)).build();
    }

    private Object buildResponseBody(Map<String, Object> input, Context context) {
        try {
            if (input.get("action").equals("warmup")) {
                sleepMinimumDuration(((Number) input.get("minimumDurationMilliseconds")).longValue());
                String uptime = Files.readAllLines(Paths.get("/proc/uptime")).stream().collect(Collectors.joining());
                return new WarmupResponse(uptime, context.getLogStreamName());
            }
            return new Response("Go Serverless v1.x! Your function executed successfully!", input);
        } catch (Exception e) {
            LOG.error("Couldn't build Response", e);
            throw new RuntimeException(e);
        }
    }

    private void sleepMinimumDuration(long minimumDurationMilliseconds) throws InterruptedException {
        LOG.info("sleeping for " + minimumDurationMilliseconds + "ms");
        Thread.sleep(minimumDurationMilliseconds);
        LOG.info("sleeping done");
    }
}

File: dotnetcore1.0/Handler.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading;
using Amazon.Lambda.Core;
using Newtonsoft.Json;

namespace AwsDotnetCsharp
{
    public class Handler
    {
        public object HandleRequest(IDictionary<string, object> request, ILambdaContext context)
        {
            Log("received: " + string.Join(",", request.Select(entry => entry.Key + "=" + entry.Value).ToArray()));
            return BuildResponseBody(request, context);
            // Or as lambda-proxy response for API Gateway:
            // return new ApiGatewayResponse(200, JsonConvert.SerializeObject(BuildResponseBody(request, context)));
        }

        private object BuildResponseBody(IDictionary<string, object> request, ILambdaContext context)
        {
            if (request["action"].Equals("warmup"))
            {
                int minimumDurationMilliseconds = (int)(long) request["minimumDurationMilliseconds"];
                Log("sleeping for " + minimumDurationMilliseconds + "ms");
                Thread.Sleep(minimumDurationMilliseconds);
                Log("sleeping done");
                return new WarmupResponse(File.ReadAllText("/proc/uptime"), context.LogStreamName);
            }
            return new Response("Go Serverless v1.0! Your function executed successfully!", request);
        }

        private void Log(string text)
        {
            LambdaLogger.Log(DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ss.fffZ") + " " + text + "\n");
        }
    }

    public class Response
    {
        public string Message { get; }
        public IDictionary<string, object> Request { get; }

        public Response(string message, IDictionary<string, object> request)
        {
            Message = message;
            Request = request;
        }
    }

    public class WarmupResponse
    {
        [JsonProperty("uptime")]
        public string Uptime { get; }

        [JsonProperty("logStreamName")]
        public string LogStreamName { get; }

        public WarmupResponse(string uptime, string logStreamName)
        {
            Uptime = uptime;
            LogStreamName = logStreamName;
        }
    }

    public class ApiGatewayResponse
    {
        [JsonProperty("statusCode")]
        public int StatusCode { get; }

        [JsonProperty("body")]
        public string Body { get; }

        public ApiGatewayResponse(int statusCode, string body) {
            StatusCode = statusCode;
            Body = body;
        }
    }
}
